.github/workflows/gradle.yml

```diff
@@ -17,20 +17,93 @@ on:
 jobs:
   build:
 
-    runs-on: macOS-latest
-
     strategy:
       matrix:
-        java: [ 8, 11, 17 ]
+        include:
+          - java: 8
+            platform: macos-latest
+            e2e-tests: android
+          - java: 11
+            platform: macos-latest
+            e2e-tests: ios
+          - java: 17
+            platform: ubuntu-latest
+      fail-fast: false
+
+    runs-on: ${{ matrix.platform }}
 
-    name: JDK ${{ matrix.java }}
+    name: JDK ${{ matrix.java }} - ${{ matrix.platform }}
     steps:
     - uses: actions/checkout@v3
+
     - name: Set up JDK ${{ matrix.java }}
       uses: actions/setup-java@v3
       with:
         distribution: 'zulu'
         java-version: ${{ matrix.java }}
         cache: 'gradle'
+
     - name: Build with Gradle
       run: ./gradlew clean build
+
+    - name: Install Node.js
+      if: ${{ matrix.e2e-tests == 'android' ||  matrix.e2e-tests == 'ios' }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: '18'
+
+    - name: Install Appium
+      if: matrix.e2e-tests == 'android' ||  matrix.e2e-tests == 'ios'
+      run: npm install -g appium@next
+
+    - name: Install UIA2 driver
+      if: matrix.e2e-tests == 'android'
+      run: appium driver install uiautomator2
+
+    - name: Prepare Android emulator
```

Comment:
Github actions already have plugins for that. Check https://github.com/appium/appium-android-driver/blob/master/.github/workflows/functional-test.yml